osu.Game.Tests/Visual/Gameplay/TestSceneGameplaySampleTriggerSource.cs

```diff
@@ -63,33 +63,30 @@ protected override IBeatmap CreateBeatmap(RulesetInfo ruleset)
             {
                 new HitCircle
                 {
-                    HitWindows = new HitWindows(),
                     StartTime = t += spacing,
                     Samples = new[] { new HitSampleInfo(HitSampleInfo.HIT_NORMAL) }
                 },
                 new HitCircle
                 {
-                    HitWindows = new HitWindows(),
                     StartTime = t += spacing,
                     Samples = new[] { new HitSampleInfo(HitSampleInfo.HIT_WHISTLE) }
                 },
                 new HitCircle
                 {
-                    HitWindows = new HitWindows(),
                     StartTime = t += spacing,
                     Samples = new[] { new HitSampleInfo(HitSampleInfo.HIT_NORMAL, HitSampleInfo.BANK_SOFT) },
                 },
                 new HitCircle
                 {
-                    HitWindows = new HitWindows(),
                     StartTime = t += spacing,
                 },
                 new Slider
                 {
-                    HitWindows = new HitWindows(),
                     StartTime = t += spacing,
                     Path = new SliderPath(PathType.Linear, new[] { Vector2.Zero, Vector2.UnitY * 200 }),
                     Samples = new[] { new HitSampleInfo(HitSampleInfo.HIT_WHISTLE, HitSampleInfo.BANK_SOFT) },
+                    // sliders usually don't have a hit window, but add one to test the scenario of playing a nested object's sample first before the main object itself.
+                    HitWindows = new OsuHitWindows(),
```

Comment:
Rather than invent weird contraptions like this, can't this test just switch ruleset to one that would natively make more sense, i.e. taiko or mania?

Reply:
It's always been the case.

Reply:
The osu! ruleset still makes sense here though. This behaviour was added if I recall to fix samples playing in the editor incorrectly.

But I'm still confused, did this test regress and if so, when? The fix proposed here doesn't answer that question. I don't think changing the timings here without properly understanding what broke the test (likely clocks?) is a good path forward, especially when the fix is harder to understand than the original.

Reply:
> This behaviour was added if I recall to fix samples playing in the editor incorrectly

`GameplaySampleTriggerSource` is only used in mania and taiko, isn't it?

I generally agree with the 'arbitrary change' sentiment though. Would be good to know what broke this.

Reply:
Right, I was a bit confused there.

Based off the failure exception, could time just be progressing further forwards than before? If it's going to be related to clocks, that's the likely part that has changed.